lms/services/digest/_models.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from functools import cached_property
+from typing import List
+
+
+@dataclass
+class HUser:
+    """A class representing H users."""
+
+    h_userid: str
+    """The H userid like 'acct:name@lms.example.com'."""
+
+    display_name: str
+    """A human readable display name."""
+
+    email: str
+    """Email for the user."""
+
+
+@dataclass
+class HCourse:
+    """A class representing courses as groups in H."""
+
+    title: str
+    """Title for the course."""
+
+    authority_provided_id: str
+    """The course's authority provided id."""
+
+    aka: List[str]
+    """All authority provided ids which map to the course from groups etc."""
+
+    instructors: List[str]
+    """A list of H userids for users who are instructors for the course."""
+
+    annotations: List[dict] = None
+    """A list of annotations in the course."""
+
+    @cached_property
+    def learner_annotations(self):
+        """Get the annotations, but filtered to users who are learners."""
+
+        return [
+            annotation
+            for annotation in self.annotations
+            if annotation["author"]["userid"] not in self.instructors
+        ]
+
+
+@dataclass
+class Digest:
+    """A digest of activity for a given user."""
+
+    audience_user: HUser
+    """The user who is the intended recipient of the information."""
```

Comment:
Technically the audience object isn't required for the serialisation. We could omit this and pass it separately.

Interestingly at the moment our digest contents aren't dependent on the user at all, only the courses.

If this was expensive we could separate the two and do the digest serialization separately and cache it. However it's cheap as chips and this is easier.